frontend/app/src/types/modules.ts

```diff
@@ -21,6 +21,28 @@ export const ModuleEnum = z.nativeEnum(Module);
 
 export type ModuleEnum = z.infer<typeof ModuleEnum>;
 
+export const DECENTRALIZED_EXCHANGES = [
+  Module.UNISWAP,
+  Module.BALANCER,
+  Module.SUSHISWAP
+];
+
+export const DEFI_PROTOCOLS = [
+  Module.YEARN,
+  Module.YEARN_V2,
+  Module.AAVE,
+  Module.MAKERDAO_DSR,
+  Module.MAKERDAO_VAULTS,
+  Module.COMPOUND,
+  Module.UNISWAP,
+  Module.LIQUITY
+] as const;
+
```

Comment:
In places where we should only pass a defi protocol we should keep a DefiProtocol type instead of changing it to module. 

If the expectation is to pass a value of the DefiProtocol subset you should not be allowed to pass an decentralized exchange module.

```typescript
export type DefiProtocol = (typeof DEFI_PROTOCOLS)[number];
```